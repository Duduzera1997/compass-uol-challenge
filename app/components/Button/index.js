import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';

function Button({ onPress }) {
  return (
    <SC.Container>
      <SC.TouchableButton onPress={onPress} testID="component-button-press">
        <SC.Label>Visualizar Not√≠cia Completa</SC.Label>
      </SC.TouchableButton>
    </SC.Container>
  );
}

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
};

export default Button;
