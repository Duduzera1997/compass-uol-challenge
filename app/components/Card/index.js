import React from 'react';
import PropTypes from 'prop-types';
import { format } from 'date-fns';
import ptBR from 'date-fns/locale/pt-BR';
import * as SC from './styles';

function Card({ imageUrl, title, abstract, noticeDate }) {
  return (
    <SC.ContainerCard>
      <SC.CardTouchable>
        <SC.CardContent flex={1}>
          <SC.PreviewImage source={{ uri: imageUrl }} />
        </SC.CardContent>
        <SC.CardContent flex={2.5}>
          <SC.Text size={15} fontBold>
            {title}
          </SC.Text>
          <SC.TextDate size={11.5} fontBold>
            {format(new Date(noticeDate), "dd 'de' MMMM 'de' yyyy", {
              locale: ptBR,
            })}
          </SC.TextDate>
          <SC.Text numberOfLines={2} size={13}>
            {abstract}
          </SC.Text>
        </SC.CardContent>
      </SC.CardTouchable>
    </SC.ContainerCard>
  );
}

Card.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  abstract: PropTypes.string.isRequired,
  noticeDate: PropTypes.string.isRequired,
};

export default Card;
