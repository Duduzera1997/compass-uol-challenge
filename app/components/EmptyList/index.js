import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';

function EmptyList({ isLoading }) {
  return (
    <SC.ContainerEmptyList>
      {isLoading && <SC.ActivityIndicator />}
      <SC.Label>
        {isLoading
          ? 'Buscando...'
          : 'Ooops! ðŸ˜ž\n Nada foi encontrado, tente novamente...'}
      </SC.Label>
    </SC.ContainerEmptyList>
  );
}

EmptyList.defaultProps = {
  isLoading: false,
};

EmptyList.propTypes = {
  isLoading: PropTypes.bool,
};

export default EmptyList;
