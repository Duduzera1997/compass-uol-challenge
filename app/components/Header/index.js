import PropTypes from 'prop-types';
import React from 'react';

import * as SC from './styles';
import { formatDateLiteral } from '~/utils';

function Header({ headerText, dateText }) {
  return (
    <SC.ContainerHeader>
      <SC.HeaderTitle>{headerText} News</SC.HeaderTitle>
      <SC.DateLabel>Hoje Ã©, {dateText}.</SC.DateLabel>
    </SC.ContainerHeader>
  );
}

Header.defaultProps = {
  dateText: formatDateLiteral(new Date()),
};

Header.propTypes = {
  headerText: PropTypes.string.isRequired,
  dateText: PropTypes.string,
};

export default Header;
