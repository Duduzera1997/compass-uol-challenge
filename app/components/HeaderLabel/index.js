import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

function HeaderLabel({ label }) {
  return (
    <View>
      <Text>{label} News</Text>
    </View>
  );
}

HeaderLabel.defaultProps = {
  label: 'Best',
};

HeaderLabel.propTypes = {
  label: PropTypes.string,
};

export default HeaderLabel;
