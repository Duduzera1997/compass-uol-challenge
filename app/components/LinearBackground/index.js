import LinearGradiend from 'react-native-linear-gradient';
import styled from 'styled-components/native';

import { darkPurple, red } from '~/styles/colors';

export default styled(LinearGradiend).attrs({
  colors: [darkPurple, red],
  start: { x: 0, y: 0 },
  end: { x: 0, y: 1.05 },
})`
  flex: 1;
`;
