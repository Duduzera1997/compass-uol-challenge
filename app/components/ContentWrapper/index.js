import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';

import Header from '../Header';
import LinearBackground from '../LinearBackground';
import SearchInput from '../SearchInput';
import ViewArticleModal from '../ViewArticleModal';

function ContentWrapper({
  title,
  children,
  showSearchInput,
  onSeachInputTextChange,
  modalProps: { visible, item, onRequestClose },
}) {
  return (
    <LinearBackground>
      <SC.Container>
        <ViewArticleModal
          visible={visible}
          item={item}
          onRequestClose={onRequestClose}
        />
        <SC.ContainerByFlex flex={1}>
          <SC.ContainerByFlex flex={1}>
            <Header headerText={title} />
          </SC.ContainerByFlex>
          {showSearchInput && (
            <SC.ContainerByFlex flex={1}>
              <SearchInput onChangeText={onSeachInputTextChange} />
            </SC.ContainerByFlex>
          )}
          <SC.ContainerByFlex flex={6}>{children}</SC.ContainerByFlex>
        </SC.ContainerByFlex>
      </SC.Container>
    </LinearBackground>
  );
}

ContentWrapper.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  showSearchInput: PropTypes.bool.isRequired,
  onSeachInputTextChange: PropTypes.func.isRequired,
  modalProps: PropTypes.shape({
    visible: PropTypes.bool.isRequired,
    item: PropTypes.shape({}),
    onRequestClose: PropTypes.func,
  }).isRequired,
};

export default ContentWrapper;
