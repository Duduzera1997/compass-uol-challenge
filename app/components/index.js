import TabIcon from './TabIcon';
import LinearBackground from './LinearBackground';
import Loading from './Loading';
import Header from './Header';
import Card from './Card';
import ListHelper from './ListHelper';
import ListArticles from './ListArticles';
import ViewArticleModal from './ViewArticleModal';
import Button from './Button';
import SearchInput from './SearchInput';
import ContentWrapper from './ContentWrapper';

export {
  TabIcon,
  LinearBackground,
  Loading,
  Header,
  Card,
  ListHelper,
  ListArticles,
  ViewArticleModal,
  Button,
  SearchInput,
  ContentWrapper,
};
