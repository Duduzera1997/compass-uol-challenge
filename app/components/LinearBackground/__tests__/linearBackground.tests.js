import React from 'react';
import { render } from '@testing-library/react-native';
import { LinearBackground } from '~/components';

describe('Linear Background', () => {
  it('should be render a linear gradient component correctly', () => {
    const { getByTestId, toJSON } = render(
      <LinearBackground testID="component-linear-background" />,
    );
    expect(getByTestId('component-linear-background')).toBeDefined();
    expect(toJSON()).toMatchSnapshot();
  });
});
