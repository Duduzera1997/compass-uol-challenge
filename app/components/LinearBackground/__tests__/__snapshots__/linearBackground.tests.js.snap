// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Linear Background should be render a linear gradient component correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4282066552,
      4294924895,
    ]
  }
  endPoint={
    Object {
      "x": 0,
      "y": 1.05,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="component-linear-background"
/>
`;
